--- a/include/v8-template.h
+++ b/include/v8-template.h
@@ -772,7 +772,8 @@ struct NamedPropertyHandlerConfiguration {
   V8_DEPRECATED("Use NamedPropertySetterCallbackV2 setter instead")
   static NamedPropertySetterCallbackV2 ConvertSetter(
       NamedPropertySetterCallback value) {
-    return NamedPropertySetterCallbackV2(value);
+    return reinterpret_cast<NamedPropertySetterCallbackV2>(
+        reinterpret_cast<void*>(value));
   }
 
   static NamedPropertyDefinerCallbackV2 ConvertDefiner(std::nullptr_t) {
@@ -785,7 +786,8 @@ struct NamedPropertyHandlerConfiguration {
   V8_DEPRECATED("Use NamedPropertyDefinerCallbackV2 definer instead")
   static NamedPropertyDefinerCallbackV2 ConvertDefiner(
       NamedPropertyDefinerCallback value) {
-    return NamedPropertyDefinerCallbackV2(value);
+    return reinterpret_cast<NamedPropertyDefinerCallbackV2>(
+        reinterpret_cast<void*>(value));
   }
 
  public:
@@ -903,7 +905,8 @@ struct IndexedPropertyHandlerConfiguration {
   V8_DEPRECATED("Use IndexedPropertySetterCallback setter instead")
   static IndexedPropertySetterCallback ConvertSetter(
       IndexedPropertySetterCallbackV2 value) {
-    return IndexedPropertySetterCallback(value);
+    return reinterpret_cast<IndexedPropertySetterCallback>(
+        reinterpret_cast<void*>(value));
   }
 
   static IndexedPropertyDefinerCallback ConvertDefiner(std::nullptr_t) {
@@ -916,7 +919,8 @@ struct IndexedPropertyHandlerConfiguration {
   V8_DEPRECATED("Use IndexedPropertyDefinerCallback definer instead")
   static IndexedPropertyDefinerCallback ConvertDefiner(
       IndexedPropertyDefinerCallbackV2 value) {
-    return IndexedPropertyDefinerCallback(value);
+    return reinterpret_cast<IndexedPropertyDefinerCallback>(
+        reinterpret_cast<void*>(value));
   }
 
  public:
