--- a/include/v8-object.h
+++ b/include/v8-object.h
@@ -5,6 +5,8 @@
 #ifndef INCLUDE_V8_OBJECT_H_
 #define INCLUDE_V8_OBJECT_H_

+#include <cstddef>
+
 #include "cppgc/garbage-collected.h"
 #include "cppgc/name-provider.h"
 #include "v8-internal.h"           // NOLINT(build/include_directory)
@@ -18,6 +20,8 @@

 namespace v8 {

+using std::nullptr_t;
+
 class Array;
 class Function;
 class FunctionTemplate;
